### PR TITLE
Add membership related features

### LINKED ISSUE #77: Retrieve for a User all his groups
Hi,
After a quick overview of the Java Client I was wondering if the API endpoint to retrieve the membership per user is available in the java Client:

https://developers.box.com/docs/#groups-get-all-group-memberships-for-a-user

Because I was not able to find it in BoxUser class.

Cheers

### PR DESCRIPTION
**Pull request by sohtsuka**
Closes #77.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @sohtsuka has signed the CLA. Thanks for the pull request!

File: src/main/java/com/box/sdk/BoxGroupMembership.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+         * @param  jsonObject the parsed JSON object.
+         */
+        Info(JsonObject jsonObject) {
+            super(jsonObject);
+        }
+
+        /**
+         * Gets the user belonging to the group.
+         *
+         * <p>Note: Only id, name and login fields are available on the initial load.</p>
```
Comment (gcurtis) [inline src/main/java/com/box/sdk/BoxGroupMembership.java]: Good thinking documenting this. Can we just rephrase this comment (as well as the others) to say something like:

"Note: the BoxUser.Info returned by this method will only have the ID, name, and login fields populated."

**Comment by gcurtis** [comment]
I've been trying to figure out why `updateMembershipSendsCorrectRequestAndParsesResponseCorrectly()` is failing. When I run it by itself with:

    gradle test --tests *updateMembershipSendsCorrectRequestAndParsesResponseCorrectly

it passes, but when it's run with the other BoxGroupMembership tests, it fails. I'm wondering if this is an issue with WireMock.

If you can't figure what's wrong, then I'm okay with just deleting that test since you also wrote an integration test that covers updating info.

**Comment by sohtsuka** [comment]
Thanks for your comments. 
As I mentioned on the pull request #100, I have been getting different errors since before this commit. All the tests pass if I comment out the following line in BoxAPIRequest.java (line 345):
```
connection.setFixedLengthStreamingMode((int) this.bodyLength);
```
but I can't figure out why the errors pop up.  I will just delete `updateMembershipSendsCorrectRequestAndParsesResponseCorrectly()`.

**Comment by gcurtis** [comment]
Thanks for the contribution!

**Pull request merged**